src/Streamly/Prelude.hs

```diff
@@ -361,6 +379,20 @@ notElem e m = go (toStream m)
 length :: (Streaming t, Monad m) => t m a -> m Int
 length = foldl (\n _ -> n + 1) 0 id
 
+-- | Returns the elements of the stream in reverse order. 
+-- The stream must be finite.
+reverse :: (Streaming t) => t m a -> t m a
+reverse m = fromStream $ go Nothing (toStream m)
+    where
+      go rev rest = Stream $ \s stp yld ->
```

Comment:
We can call the `Svar` as `svr` just like `stp` and `yld`, that will provide a bit more hint about what it is. Historically I called it `ctx` everywhere but we can change that now.